pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -271,11 +271,12 @@ func (s *DiscoveryServer) updateServiceShards(push *model.PushContext) error {
 	for _, svc := range push.Services(nil) {
 		for _, registry := range nonK8sRegistries {
 			// in case this svc does not belong to the registry
-			if svc, _ := registry.GetService(svc.Hostname); svc == nil {
+			if svc.Attributes.ServiceRegistry != string(registry.Provider()) {
+				adsLog.Warnf("skipping service %s for provider %s, service registry", svc.Hostname, registry.Provider(), svc.Attributes.ServiceRegistry)
```

Comment:
This is not a warning

Reply:
I know :-) Just put it for debug as the tests are failing only CI. Will remove once tests are passed